### PR TITLE
feat: add target tokens distribution

### PR DESCRIPTION
**Pull request by nchamo**
We are now making the tokens distribution configurable. The contract owner (and other allowed users) will be able to set it

### CODE REVIEW

File: contracts/DCAFeeManager/DCAFeeManager.sol [hunk 1]
```diff
@@ -12,7 +12,50 @@ contract DCAFeeManager is Governable, IDCAFeeManager {
   /// @inheritdoc IDCAFeeManager
   IDCAHub public immutable hub;
 
-  constructor(IDCAHub _hub, address _governor) Governable(_governor) {
+  TargetTokenShare[] internal _distribution;
+
+  constructor(
+    IDCAHub _hub,
+    TargetTokenShare[] memory _distributionToSet,
```
Comment (0xged) [inline contracts/DCAFeeManager/DCAFeeManager.sol:19]: is `calldata` not allowed in constructor?

File: contracts/DCAFeeManager/DCAFeeManager.sol [hunk 2]
```diff
@@ -12,7 +12,50 @@ contract DCAFeeManager is Governable, IDCAFeeManager {
...
+    address _governor
+  ) Governable(_governor) {
     hub = _hub;
+    setTargetTokensDistribution(_distributionToSet);
+  }
+
+  /// @inheritdoc IDCAFeeManager
+  function targetTokensDistribution() external view returns (TargetTokenShare[] memory) {
+    return _distribution;
+  }
```
Comment (0xged) [inline contracts/DCAFeeManager/DCAFeeManager.sol:29]: it didnt allow you to make the variable public? 😢

File: contracts/DCAFeeManager/DCAFeeManager.sol [hunk 3]
```diff
@@ -12,7 +12,50 @@ contract DCAFeeManager is Governable, IDCAFeeManager {
...
+      for (uint256 i = _min; i < _newDistribution.length; i++) {
+        _assignedShares += _newDistribution[i].shares;
+        _distribution.push(_newDistribution[i]);
+      }
+    } else if (_currentTargetTokens > _newDistribution.length) {
+      // If have less tokens than before, then remove extra tokens
+      for (uint256 i = _min; i < _currentTargetTokens; i++) {
+        _distribution.pop();
+      }
+    }
```
Comment (0xged) [inline contracts/DCAFeeManager/DCAFeeManager.sol:56]: I think you can avoid re-declaring `i` here. If you declare `i` out of the first `for` scope, then at the end of the loop `i = min`, right? and you can avoid the re-declaration.

**Review by 0xged — APPROVED** [review]
other than the comments, that are nit, this looks good!

File: contracts/DCAFeeManager/DCAFeeManager.sol (hunk 1 above)
Comment (nchamo) [reply contracts/DCAFeeManager/DCAFeeManager.sol:19]: No, must be memory. Not sure why

File: contracts/DCAFeeManager/DCAFeeManager.sol (hunk 2 above)
Comment (nchamo) [reply contracts/DCAFeeManager/DCAFeeManager.sol:29]: I can make it public, but the getter will be by index, not the entire array. That's why I did it like this

File: contracts/DCAFeeManager/DCAFeeManager.sol (hunk 3 above)
Comment (nchamo) [reply contracts/DCAFeeManager/DCAFeeManager.sol:56]: Great catch!

**Pull request merged**